internal/controller/volumereplicationgroup_controller.go

```diff
@@ -125,6 +125,8 @@ func (r *VolumeReplicationGroupReconciler) SetupWithManager(
 		r.Log.Info("Kube object protection disabled; don't watch kube objects requests")
 	}
 
+	r.recipeStatus = make(map[string]*util.RecipeStatus)
```

Comment:
Good catch. I wonder how long it has been there.

Reply:
Why initialize here and not when we create the object? 

We have 2 options:
- initialize when we create the object
- initialize on the first use

The first option should be the default unless we have a good reason to have more complicate code.

Reply:
The object is created [here](https://github.com/RamenDR/ramen/blob/10cd07136e99a6663bcfd560ce739be29487e583/cmd/main.go#L170-L176) and this is a private member of the structure (like `veleroCRsAreWatched`) and cannot be initialized there. The function SetupWithManager is invoked to complete the setup of the object, hence its location in this function seems apt.

Reply:
Also, cannot be on first use, as the reconcile is multi-threaded, hence in a racy case there maybe 2 routines that may initialize this global.